ui/raidboss/data/06-ew/raid/p4s.ts

```diff
@@ -142,6 +185,195 @@ const triggerSet: TriggerSet<Data> = {
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['69DE', '69DF', '69E0', '69E1'], source: 'Hesperos' }),
+      netRegexDe: NetRegexes.startsUsing({ id: ['69DE', '69DF', '69E0', '69E1'], source: 'Hesperos' }),
+      netRegexFr: NetRegexes.startsUsing({ id: ['69DE', '69DF', '69E0', '69E1'], source: 'Hespéros' }),
+      netRegexJa: NetRegexes.startsUsing({ id: ['69DE', '69DF', '69E0', '69E1'], source: 'ヘスペロス' }),
+      preRun: (data) => {
+        if (!data.beloneCoilsTwo) {
+          delete data.debuffRole;
+          delete data.tetherRole;
+          data.hasRoleCall = false;
+          data.ignoreChlamys = true;
+        } else {
+          data.ignoreChlamys = false;
+        }
+      },
+      suppressSeconds: 1,
+      response: (data, matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = roleOutputStrings;
+
+        const roles: { [role: string]: string } = {
+          'dps': output.dps!(),
+          'tank/healer': output.tankHealer!(),
+        };
+
+        const roleTowers = matches.id === '69DE' ? 'dps' : 'tank/healer';
+        const roleOther = matches.id === '69DE' ? 'tank/healer' : 'dps';
+
+        // Second Coils = Debuffs later
+        if (data.beloneCoilsTwo) {
+          if (roleTowers === 'dps') {
+            (data.debuffRole ??= []).push('healer');
+            data.debuffRole.push('tank');
+          } else {
+            (data.debuffRole ??= []).push('dps');
+          }
+
+          // For second coils, if you are not in the debuff list here you are tower
+          if (!data.debuffRole.includes(data.role))
+            return { ['alertText']: output.roleTowers!({ role: roles[roleTowers] }) };
+
+          // If you have tethers and debuff, you need everything
+          const tetherRole = data.tetherRole ??= [];
+          const debuffRole = data.debuffRole ??= [];
+          if (debuffRole[0] === tetherRole[0])
+            return { ['infoText']: output.roleEverything!({ role: roles[roleOther] }) };
+          return { ['infoText']: output.roleDebuffs!({ role: roles[roleOther] }) };
+        }
+
+        // First Coils = Tethers later
+        if (roleTowers === 'dps') {
+          (data.tetherRole ??= []).push('healer');
+          data.tetherRole.push('tank');
+        } else {
+          (data.tetherRole ??= []).push('dps');
+        }
+
+        // For first coils, there are tower and tethers
+        if (data.tetherRole.includes(data.role))
+          return { ['alertText']: output.roleTethers!({ role: roles[roleOther] }) };
+        return { ['alertText']: output.roleTowers!({ role: roles[roleTowers] }) };
+      },
+      run: (data) => data.beloneCoilsTwo = true,
+    },
+    {
+      id: 'P4S Role Call',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: ['AF2', 'AF3'], capture: true }),
+      condition: Conditions.targetIsYou(),
+      alertText: (data, matches, output) => {
```

Comment:
```suggestion
      infoText: (data, matches, output) => {
```

I think better to differentiate pass severity from get, especially since the "stack up" strategy means the "pass" people don't do anything.